ui/imodel-components-react/src/imodel-components-react/quantityformat/v2/FormatPanelV2.tsx

```diff
@@ -0,0 +1,137 @@
+} from "./panels/Fractional.js";
+import {
+  getScientificPrimaryChildren,
+  getScientificSecondaryChildren,
+} from "./panels/Scientific.js";
+import {
+  getStationPrimaryChildren,
+  getStationSecondaryChildren,
+} from "./panels/Station.js";
+import {
+  getAzimuthPrimaryChildren,
+  getAzimuthSecondaryChildren,
+} from "./panels/Azimuth.js";
+import {
+  getBearingPrimaryChildren,
+  getBearingSecondaryChildren,
+} from "./panels/Bearing.js";
+import "./FormatPanelV2.scss";
+import {
+  getRatioPrimaryChildren,
+  getRatioSecondaryChildren,
+} from "./panels/Ratio.js";
+import { useTranslation } from "../../useTranslation.js";
+
+/** Props for FormatPanelV2
+ * @internal
+ */
+export interface FormatPanelV2Props {
+  formatProps: FormatProps;
+  unitsProvider: UnitsProvider;
+  onFormatChange: (formatProps: FormatProps) => void;
+  persistenceUnit?: UnitProps;
+}
+
+/** Format Panel V2 that uses primary and secondary children based on format type
+ * @beta
+ */
+export function FormatPanelV2(props: FormatPanelV2Props) {
+  const { formatProps, unitsProvider, onFormatChange, persistenceUnit } = props;
+  const [isExpanded, setIsExpanded] = React.useState(false);
+  const { translate } = useTranslation();
+
+  const [primaryChildren, secondaryChildren] = React.useMemo(() => {
+    const panelProps: PanelProps = {
+      formatProps,
+      unitsProvider,
+      onFormatChange,
+      persistenceUnit,
+    };
+    const formatType = parseFormatType(formatProps.type, "format");
+
+    switch (formatType) {
+      case FormatType.Decimal:
+        return [
+          getDecimalPrimaryChildren(panelProps),
+          getDecimalSecondaryChildren(panelProps),
+        ];
+      case FormatType.Fractional:
+        return [
+          getFractionalPrimaryChildren(panelProps),
+          getFractionalSecondaryChildren(panelProps),
+        ];
+      case FormatType.Scientific:
+        return [
+          getScientificPrimaryChildren(panelProps),
+          getScientificSecondaryChildren(panelProps),
+        ];
+      case FormatType.Station:
+        return [
+          getStationPrimaryChildren(panelProps),
+          getStationSecondaryChildren(panelProps),
+        ];
+      case FormatType.Azimuth:
+        return [
+          getAzimuthPrimaryChildren(panelProps),
+          getAzimuthSecondaryChildren(panelProps),
+        ];
+      case FormatType.Bearing:
+        return [
+          getBearingPrimaryChildren(panelProps),
+          getBearingSecondaryChildren(panelProps),
+        ];
+      case FormatType.Ratio:
+        return [
+          getRatioPrimaryChildren(panelProps),
+          getRatioSecondaryChildren(panelProps),
+        ];
+      default:
+        return [
+          getDecimalPrimaryChildren(panelProps),
+          getDecimalSecondaryChildren(panelProps),
+        ];
+    }
+  }, [formatProps, unitsProvider, onFormatChange, persistenceUnit]);
+
+  return (
+    <div className="format-panel-v2">
```

Comment:
Could probably use the [iTwinUI flex utility](https://itwinui.bentley.com/docs/flex) instead.